var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -310,7 +310,7 @@ def bootstrap_args(self):
         if not sys.platform == "win32":
             args.append("--prefix={0}".format(self.prefix))
 
-            if spack.build_environment.should_set_parallel_jobs(jobserver_support=True):
+            if not spack.build_environment.jobserver_enabled():
```

Comment:
This was actually the reason there was a `should_set_parallel_jobs` function: the idea was to have it return true if and only if `-j` should be passed explicitly.

Reply:
I consciously dropped the support of `SPACK_NO_PARALLEL_MAKE` here because of the `-j`/not-`-j` confusion, and it didn't seem supported before the introduction of `should_set_parallel_jobs`.

I'll work on reintroducing it.